src/tests/EShopworld.WorkerProcess.IntegrationTests/WorkerLeaseTests.cs

```diff
@@ -126,6 +126,40 @@ public async Task TestMultipleWorkLeases()
             _expiredList.Should().NotBeEmpty();
         }
 
+        [Fact, IsIntegration]
+        public async Task TestMultipleConcurrentWorkLeases()
+        {
+            ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+
+            // Arrange
+            Random r = new Random();
+            var leaseStore = _serviceProvider.GetService<ILeaseStore>();
+            SetupWorkerLeases(30, i => r.Next(7));
```

Comment:
I am a little bit uncomfortable with the random generation of the priority. Wouldn't it be better to have a fixed pattern of priorities? May be even more than one (with a theory)? @chomick what's your thought here?

Reply:
My point was, no matter what the priority we choose, the winner should always be the minimum, hence the random generation. But i'm open to alternatives of course

Reply:
If you leave it for random selection, your tests will be less consistent. Agree with Nino that having priorities spread out deterministically would be better

Reply:
Ok, I don't fully agree on this, but i'm totally fine for an alternative, e.g using a decresing priority number strategy like MaxPriority - i % MaxPriority?

Reply:
changed priority strategy to a deterministic one